opal/models.py

```diff
@@ -1662,6 +1690,35 @@ def explicit_access_only(self):
         return any(r for r in all_roles if r == "scientist")
 
 
+class Stage(TrackedModel):
+    """ An episode can have many stages, but should have no more than
+        one with no stop date any time.
+
+        This is expected to be set by the episode_category
+        which determines what stages are available to an episode.
+    """
+    episode = models.ForeignKey(Episode, null=False)
+    started = models.DateTimeField()
+    stopped = models.DateTimeField(blank=True, null=True)
+    value = models.CharField(max_length=256)
+
+    @transaction.atomic
+    def save(self, *args, **kwargs):
+        """ An episode should not have multiple open stages.
+            raise exceptions and roll back if anyone tries
+            any funny business.
+        """
+        if not self.id:
+            if(self.episode.stage_set.filter(stopped=None).exists()):
+                if not self.stopped:
+                    err = "for episode {}, stage {}. An episode cannot have \
```

Comment:
Can we look at this error message again? 'For x,y` isn't a regular phrasing, and I think they should start with sentence case...